File: sdkwork-ai-sdk-java/src/main/java/com/sdkwork/ai/api/AssistantApi.java
package com.sdkwork.ai.api;

import com.sdkwork.ai.http.HttpClient;
import com.sdkwork.ai.model.*;
import java.util.List;
import java.util.Map;

public class AssistantApi {
    private final HttpClient client;
    
    public AssistantApi(HttpClient client) {
        this.client = client;
    }

    /** Get assistant */
    public AssistantResponse getAssistant(String assistant_id) throws Exception {
        return (AssistantResponse) client.get(ApiPaths.aiPath("/assistants/" + assistant_id + ""));
    }

    /** Update assistant */
    public AssistantResponse updateAssistant(String assistant_id, AssistantCreateRequest body) throws Exception {
        return (AssistantResponse) client.post(ApiPaths.aiPath("/assistants/" + assistant_id + ""), body);
    }

    /** Delete assistant */
    public AssistantDeleteResponse deleteAssistant(String assistant_id) throws Exception {
        return (AssistantDeleteResponse) client.delete(ApiPaths.aiPath("/assistants/" + assistant_id + ""));
    }

    /** List assistants */
    public AssistantListResponse listAssistants(Map<String, Object> params) throws Exception {
        return (AssistantListResponse) client.get(ApiPaths.aiPath("/assistants"), params);
    }

    /** Create assistant */
    public AssistantResponse createAssistant(AssistantCreateRequest body) throws Exception {
        return (AssistantResponse) client.post(ApiPaths.aiPath("/assistants"), body);
    }
}

File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/api/AssistantApi.kt
package com.sdkwork.ai.api

import com.sdkwork.ai.*
import com.sdkwork.ai.http.HttpClient

class AssistantApi(private val client: HttpClient) {

    /** Get assistant */
    suspend fun getAssistant(assistant_id: String): AssistantResponse? {
        return client.get(ApiPaths.aiPath("/assistants/$assistant_id")) as? AssistantResponse
    }

    /** Update assistant */
    suspend fun updateAssistant(assistant_id: String, body: AssistantCreateRequest): AssistantResponse? {
        return client.post(ApiPaths.aiPath("/assistants/$assistant_id"), body) as? AssistantResponse
    }

    /** Delete assistant */
    suspend fun deleteAssistant(assistant_id: String): AssistantDeleteResponse? {
        return client.delete(ApiPaths.aiPath("/assistants/$assistant_id")) as? AssistantDeleteResponse
    }

    /** List assistants */
    suspend fun listAssistants(params: Map<String, Any>? = null): AssistantListResponse? {
        return client.get(ApiPaths.aiPath("/assistants"), params) as? AssistantListResponse
    }

    /** Create assistant */
    suspend fun createAssistant(body: AssistantCreateRequest): AssistantResponse? {
        return client.post(ApiPaths.aiPath("/assistants"), body) as? AssistantResponse
    }
}
